torchvision/models/video/s3d.py

```diff
@@ -0,0 +1,173 @@
+        )
+        self.branch2 = nn.Sequential(
+            Conv3dNormActivation(
+                in_planes, b2_mid, kernel_size=1, stride=1, norm_layer=partial(nn.BatchNorm3d, eps=1e-3, momentum=1e-3)
+            ),
+            SepConv3d(b2_mid, b2_out, kernel_size=3, stride=1, padding=1),
+        )
+        self.branch3 = nn.Sequential(
+            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=1, padding=1),
+            Conv3dNormActivation(
+                in_planes, b3_out, kernel_size=1, stride=1, norm_layer=partial(nn.BatchNorm3d, eps=1e-3, momentum=1e-3)
+            ),
+        )
+
+    def forward(self, x):
+        x0 = self.branch0(x)
+        x1 = self.branch1(x)
+        x2 = self.branch2(x)
+        x3 = self.branch3(x)
+        out = torch.cat((x0, x1, x2, x3), 1)
+
+        return out
+
+
+class S3D(nn.Module):
+    """S3D main class.
+
+    Args:
+        num_class (int): number of classes for the classification task.
+
+    Inputs:
+        x (Tensor): batch of videos with dimensions (batch, channel, time, height, width)
+    """
+
+    def __init__(
+        self,
+        num_classes: int = 400,
+    ) -> None:
+        super().__init__()
+        self.base = nn.Sequential(
+            SepConv3d(3, 64, kernel_size=7, stride=2, padding=3),
+            nn.MaxPool3d(kernel_size=(1, 3, 3), stride=(1, 2, 2), padding=(0, 1, 1)),
+            Conv3dNormActivation(
+                64, 64, kernel_size=1, stride=1, norm_layer=partial(nn.BatchNorm3d, eps=1e-3, momentum=1e-3)
+            ),
+            SepConv3d(64, 192, kernel_size=3, stride=1, padding=1),
+            nn.MaxPool3d(kernel_size=(1, 3, 3), stride=(1, 2, 2), padding=(0, 1, 1)),
+            SepInceptionBlock3D(192, [[64], [96, 128], [16, 32], [32]]),
+            SepInceptionBlock3D(256, [[128], [128, 192], [32, 96], [64]]),
+            nn.MaxPool3d(kernel_size=(3, 3, 3), stride=(2, 2, 2), padding=(1, 1, 1)),
+            SepInceptionBlock3D(480, [[192], [96, 208], [16, 48], [64]]),
+            SepInceptionBlock3D(512, [[160], [112, 224], [24, 64], [64]]),
+            SepInceptionBlock3D(512, [[128], [128, 256], [24, 64], [64]]),
+            SepInceptionBlock3D(512, [[112], [144, 288], [32, 64], [64]]),
+            SepInceptionBlock3D(528, [[256], [160, 320], [32, 128], [128]]),
+            nn.MaxPool3d(kernel_size=(2, 2, 2), stride=(2, 2, 2), padding=(0, 0, 0)),
+            SepInceptionBlock3D(832, [[256], [160, 320], [32, 128], [128]]),
+            SepInceptionBlock3D(832, [[384], [192, 384], [48, 128], [128]]),
+        )
+        self.fc = nn.Conv3d(1024, num_classes, kernel_size=1, stride=1, bias=True)
+
+    def forward(self, x):
+        y = self.base(x)
+        y = F.avg_pool3d(y, (2, y.size(3), y.size(4)), stride=1)
+        y = self.fc(y)
+        y = y.view(y.size(0), y.size(1), y.size(2))
+        logits = torch.mean(y, 2)
+
+        return logits
+
+
+class S3D_Weights(WeightsEnum):
+    pass
```

Comment:
Again once everything is inplace, we would need to offer weights. Let's leave this last. Are there official paper weights we can use?

Reply:
I used [these weights](https://github.com/kylemin/S3D#weight-file--sample-code) pretrained on kinetics400. they were used in the project i onboarded to torchmultimodal, but they're not the official weights (which don't exist as far as i know) from the paper

Reply:
It's not clear from the repo what is the size of the video they use. I don't see any resize statement so I assume it's `384 × 256 px` as the images on the repo. The specific aspect ratio is a bit odd/non-standard. As part of adding the weights, we would have to verify the accuracy on Kinetics400 and guesstimate the central-crop/resize they are using to reproduce the results.